[ch112] Landing page for Data Inputter

seerocode opened the pull request:
Story details: https://app.clubhouse.io/stanford-computational-policy-lab/story/112

How she's lookin:
![image](https://user-images.githubusercontent.com/10480484/113021326-766a7980-9151-11eb-94d1-593d8d58abe8.png)

Commit:
feat(ch112): add search bar for homepage
- chore: Add utility file with function to capitalize first letter
- test: Add test for search bar

Commit:
feat(ch112): add table of datasets
- test: add a11y test for datasets table

Commit:
feat(ch112): add user homepage component

Commit:
refactor(ch112): refactor app function  and router

Commit:
chore(ch112): upgrade yarn dependencies

Commit:
chore(ch112): Fix linting issues

Commit:
test(ch112): fix snapshot test for app

Commit:
test(ch112): Mock i18next and update search data prop

Commit:
chore(ch112): cleanup import for deleted css

jnu commented on client/src/components/SearchAutoComplete.tsx:
```diff
@@ -0,0 +1,33 @@
+        option!.value.toUpperCase().indexOf(inputValue.toUpperCase()) !== -1
+      }
+      aria-expanded="false"
+      notFoundContent={`${capitalizeFirstLetter(t("program"))} not found`}
```
hm, i wonder if `toUpperCase` actually does the right thing for all languages / alphabets? not worth dwelling on, but maybe check first to see whether this i18n logic has any helper methods built in for cross-language capitalization

seerocode replied:
I ended up removing the function and replacing it with a css selector: https://developer.mozilla.org/en-US/docs/Web/CSS/::first-letter

jnu reviewed (approved):
looks good! great work!! as for the tests, let's look at those together tomorrow and see if there's a way we can fix them without rewriting an entirely new component. thanks!!

Commit:
style(ch112): lint tslint file

Commit:
test(ch112): ignore empty block for i18n mock

Commit:
style(ch112): lint app and search files

Commit:
style(ch112): capitalize first letter w css
- Remove util to capitalize first letter
- Use ::first-letter css selector capitalize first letter of search not found

Commit:
style(ch112): lint search test

Commit:
feat(ch112): change filter option for search

Commit:
test(ch112): skip a11y test - inconclusive
